src/unlimited.cpp

```diff
@@ -1026,6 +1027,57 @@ bool CheckExcessive(const CBlock &block, uint64_t blockSize, uint64_t nSigOps, u
     return false;
 }
 
+bool CheckDynamic(const CBlock &block, uint64_t blockSize, uint64_t nSigOps, uint64_t nTx, uint64_t largestTx)
+{
+    if (blockSize > sizeTracker.GetMaxBlockSize())
+    {
+        LOGA("Excessive block: Rejecting block with ver:%x time:%d size: %" PRIu64 " Tx:%" PRIu64 " Sig:%d  :too many bytes\n",
+            block.nVersion, block.nTime, blockSize, nTx, nSigOps);
+        return false;
+    }
+
+    if (blockSize > BLOCKSTREAM_CORE_MAX_BLOCK_SIZE)
+    {
+        // Check transaction size to limit sighash
+        if (largestTx > maxTxSize.Value())
+        {
+            LOGA("Excessive block: ver:%x time:%d size: %" PRIu64 " Tx:%" PRIu64
+                 " largest TX:%d  :tx too large.  Expected less than: %d\n",
+                block.nVersion, block.nTime, blockSize, nTx, largestTx, maxTxSize.Value());
+            return false;
+        }
+
+        // check proportional sigops
+        uint64_t blockMbSize =
+            1 + ((blockSize - 1) /
+                    1000000); // block size in megabytes rounded up. 1-1000000 -> 1, 1000001-2000000 -> 2, etc.
```

Comment:
this formula seems to be wrong. Even if the blockSize is below 1mb, it always returns at least 1Mb

Reply:
shouldn't it be as simple as:

`uint64_t blockMbSize = blockSize / 1000000`

You'd only get the size in mb as an integer though.
If you had a 2.5Mb block, say in decimal mbs out of `2500000 bytes`, then the result would be rounded down to `2 Mbs` due to integer division.

Reply:
that code actually isnt from this PR. that function is a duplicate of CheckExcessive for basic outlining purposes. this PR is more or less permanently on hold until the full spec is ironed out and scheduled into a HF